owner/manager.go

```diff
@@ -179,6 +185,32 @@ func (m *ownerManager) CampaignOwner(ctx context.Context) error {
 	return nil
 }
 
+// ResignOwner lets the owner start a new election.
+func (m *ownerManager) ResignOwner(ctx context.Context) error {
+	elec := (*concurrency.Election)(atomic.LoadPointer(&m.elec))
+	if elec == nil {
+		return errors.Errorf("This node is not a ddl owner, can't be resigned.")
+	}
+
+	err := elec.Resign(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	log.Warnf("%s Resign ddl owner success!", m.logPrefix)
+	return nil
+}
+
+func (m *ownerManager) toBeOwner(elec *concurrency.Election) {
+	atomic.StorePointer(&m.elec, unsafe.Pointer(elec))
+	m.SetOwner(true)
+}
+
+func (m *ownerManager) retireOwner() {
+	m.SetOwner(false)
+	atomic.StorePointer(&m.elec, nil)
```

Comment:
Could we only use the `elec` to determine if it is the owner? Then we do not need to maintain the `owner` anymore.

Reply:
There are some test cases use SetOwner(false) and just use elec to determin if it is the owner will not easy to mock in tests.

Reply:
Can we change `SetOwner(isOwner bool)` to `SetOwner(elec unsafe.Pointer)`?
In the `mockManager`, we can use any not nil pointer to represent `elec`.